packages/components/src/Icon/Icon.component.js

```diff
@@ -36,12 +36,29 @@ export const TRANSFORMS = Object.keys(FA_TRANSFORMS);
  * @example
 <Icon name="fa-bars"></Icon>
  */
-function Icon({ className, name, title, transform, onClick }) {
+function Icon({ className, name, title, transform, onClick, href }) {
 	const accessibility = {
 		focusable: 'false', // IE11
 		'aria-hidden': 'true',
 		title: title || null,
 	};
+	if (!!href) {
+		let props = {
+			src: href,
+			className: classnames(
+				theme['tc-icon'],
+				'tc-icon',
+				className
+			)
+		};
+		if (!!onClick) {
+			props.onClick = onClick;
+		}
+		if (!!title) {
+			props.title = title;
+		}
+		return <img {...props} />;
```

Comment:
Please refrain from using spread operators. Passing the props explicitly make code clearer and avoids allocation and computation.

Reply:
how can I manage the optional values here without that pattern?